lib/order.rb

```diff
@@ -0,0 +1,41 @@
+require_relative 'menu'
+
+class Order < Menu
+
+	attr_reader :order, :texted
+
+	attr_accessor :texted
+
+	def initialize
+		@order = {}
+		@menu_keys = menu_list.keys
+		@menu_values = menu_list.values
+		@texted = false
+	end
+
+	def new_order(list,total_cost)
+		count = 0
+		puts "ORDER SUMMARY"
+		list.each do |x,y|
+			dish = menu_keys[x - 1]
+			cost = menu_values[x - 1] * y
+			puts "#{(list.index([x,y]))+1}. #{dish}   x #{y}   £#{cost}"
+			order.merge!({dish => cost})
+		end
+		sum = 0
+		order.values.each {|a| sum += a }
+		fail "Total cost is incorrect" if sum != total_cost
+		puts "TOTAL COST - £#{total_cost}" 
+	end
```

Comment:
It's a little hard to follow the logic here, they're not very book-ish. These lines should be extracted to a private method.